Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp

```diff
@@ -103,68 +104,313 @@ void HTMLLinkElement::parse_attribute(DeprecatedFlyString const& name, Deprecate
+    request->set_policy_container(options.policy_container);
+
+    // 6. Set request's integrity metadata to options's integrity.
+    request->set_integrity_metadata(options.integrity);
+
+    // 7. Set request's cryptographic nonce metadata to options's cryptographic nonce metadata.
+    request->set_cryptographic_nonce_metadata(options.cryptographic_nonce_metadata);
+
+    // 8. Set request's referrer policy to options's referrer policy.
+    request->set_referrer_policy(options.referrer_policy);
+
+    // 9. Set request's client to options's environment.
+    request->set_client(options.environment);
+
+    // 10. Return request.
+    return request;
+}
+
+// https://html.spec.whatwg.org/multipage/semantics.html#fetch-and-process-the-linked-resource
+void HTMLLinkElement::fetch_and_process_linked_resource()
+{
+    default_fetch_and_process_linked_resource();
+}
+
+// https://html.spec.whatwg.org/multipage/semantics.html#default-fetch-and-process-the-linked-resource
+void HTMLLinkElement::default_fetch_and_process_linked_resource()
+{
+    // https://html.spec.whatwg.org/multipage/semantics.html#the-link-element:attr-link-href-4
+    // If both the href and imagesrcset attributes are absent, then the element does not define a link.
+    // FIXME: Support imagesrcset attribute
+    if (!has_attribute(AttributeNames::href) || href().is_empty())
+        return;
+
+    // 1. Let options be the result of creating link options from el.
+    auto options = create_link_options();
+
+    // 2. Let request be the result of creating a link request given options.
+    auto request = create_link_request(options);
+
+    // 3. If request is null, then return.
+    if (request == nullptr) {
+        return;
+    }
+
+    // FIXME: 4. Set request's synchronous flag.
+
+    // 5. Run the linked resource fetch setup steps, given el and request. If the result is false, then return.
+    if (!linked_resource_fetch_setup_steps(*request))
+        return;
+
+    // 6. Set request's initiator type to "css" if el's rel attribute contains the keyword stylesheet; "link" otherwise.
+    if (m_relationship & Relationship::Stylesheet) {
+        request->set_initiator_type(Fetch::Infrastructure::Request::InitiatorType::CSS);
     } else {
-        dbgln_if(CSS_LOADER_DEBUG, "HTMLLinkElement: Resource did load, has encoded data. URL: {}", resource()->url());
+        request->set_initiator_type(Fetch::Infrastructure::Request::InitiatorType::Link);
+    }
 
-        if (resource()->mime_type() != "text/css"sv) {
-            dbgln_if(CSS_LOADER_DEBUG, "HTMLLinkElement: Resource did load, but MIME type was {} instead of text/css. URL: {}", resource()->mime_type(), resource()->url());
-            return;
-        }
+    // 7. Fetch request with processResponseConsumeBody set to the following steps given response response and null, failure, or a byte sequence bodyBytes:
+    (void)Fetch::Fetching::fetch(
```

Comment:
This should probably not ignore errors, let's use `release_value_but_fixme_should_propagate_errors()` on the return value for better visibility